system/jlib/jstats.cpp

```diff
@@ -1988,9 +2012,69 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
             cur.visit(visitor);
     }
 
+    virtual bool refreshAggregates(std::vector<StatisticKind> & aggregateKinds, std::vector<unsigned __int64> & totals) override
+    {
+        assertex(aggregateKinds.size()==totals.size());
+
+        bool updated = false;
+        if (queryScopeType()==SSTsubgraph)
+        {
+            ForEachItemIn(i, stats)
+            {
+                Statistic & stat = stats.element(i);
+                StatisticKind kind = stat.queryKind();
+                auto iteratorVec = std::find(aggregateKinds.begin(), aggregateKinds.end(), kind);
+                if (iteratorVec!=aggregateKinds.end())
+                {
+                    unsigned pos = iteratorVec-aggregateKinds.begin();
+                    totals[pos] += stat.queryValue();
+                    updated = true;
+                }
+            }
+        }
+        else
+        {
+            std::vector<unsigned __int64> childTotals(aggregateKinds.size());
+            for (auto & child : children)
+            {
+                if (child.refreshAggregates(aggregateKinds, childTotals))
+                    updated = true;
+            }
+            if (updated)
+            {
+                updated=false;
+                std::vector<unsigned __int64>::iterator totalIter = totals.begin();
+                std::vector<unsigned __int64>::iterator subTotalIter = childTotals.begin();
+                std::vector<StatisticKind>::iterator kindIter = aggregateKinds.begin();
+                while (totalIter != totals.end())
+                {
+                    if (updateStatistic(*kindIter, *subTotalIter, StatsMergeReplace))
```

Comment:
This is StatsMergeReplace because it is updating a single value for this scope.   The value has already been "merged" at the child level.  The loop 2092-2101 generates the subTotal value by using the correct mergeAction (line 2101).  In line 2122, the one subTotal per kind is written to this scope.